gtsfm/loader/one_d_sfm_loader.py

```diff
@@ -0,0 +1,163 @@
+"""Loader for datasets used in 1DSFM and Colmap papers.
+
+Authors: Yanwei Du
+"""
+
+import glob
+import os
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+import numpy as np
+from gtsam import Cal3Bundler, Pose3
+
+import gtsfm.utils.io as io_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+
+class OneDSFMLoader(LoaderBase):
+    """Loader for datasets used in 1DSFM and Colmap papers.
+
+    The dataset is collected from internet and only partial images have exif data embedded. It can be downloaded from
+    1DSFM [project website](https://www.cs.cornell.edu/projects/1dsfm/) and has the following folder structure:
+
+        - images.SEQ_NAME/SEQ_NAME/images/*.jpg
+
+    Reference:
+    [1] Wilson, K., Snavely, N. (2014). Robust Global Translations with 1DSfM. In: Fleet, D., Pajdla, T., Schiele, B.,
+    Tuytelaars, T. (eds) Computer Vision – ECCV 2014. ECCV 2014. Lecture Notes in Computer Science, vol 8691. Springer,
+    Cham. https://doi.org/10.1007/978-3-319-10578-9_5
+    [2] J. L. Schönberger and J. -M. Frahm, "Structure-from-Motion Revisited," 2016 IEEE Conference on Computer Vision
+    and Pattern Recognition (CVPR), 2016, pp. 4104-4113, doi: 10.1109/CVPR.2016.445.
+    """
+
+    def __init__(
+        self,
+        folder: str,
+        image_extension: str = "jpg",
+        max_resolution: int = 640,
+        max_num_imgs: int = 0,
+    ) -> None:
+        """Initializes to load from a specified folder on disk.
+
+        Args:
+            folder: the base folder which contains image sequence.
+            image_extension: file extension for the image files. Defaults to 'jpg'.
+            max_resolution: integer representing maximum length of image's short side, i.e.
+                the smaller of the height/width of the image. e.g. for 1080p (1920 x 1080),
+                max_resolution would be 1080. If the image resolution max(height, width) is
+                greater than the max_resolution, it will be downsampled to match the max_resolution.
+            max_num_imgs: integer representing max number of images to process in the sequence,
+                0 or negative values means to process all the images.
+        """
+        super().__init__(max_resolution=max_resolution)
+
+        # Fetch all the file names in /images folder.
+        search_path = os.path.join(folder, "images", f"*.{image_extension}")
+
+        # NOTE(yanwei.du) Currently we only process images with valid EXIF data.
+        (
+            self._image_paths,
+            self._num_exif_imgs,
+            self._num_all_imgs,
+        ) = self.get_images_with_valid_exif(search_path, max_num_imgs)
+        self._num_imgs = len(self._image_paths)
+
+        print(f"Selected {self._num_imgs} out of {self._num_exif_imgs} images with valid exif.")
```

Comment:
if possible, let's use logger.info instead of print here, to unify the logging. We usually put something like
```python
logger = logger_utils.get_logger()
import gtsfm.utils.logger as logger_utils

logger.info("Selected %d out of %d images with valid exif.", self._num_imgs, self._num_exif_imgs)
```

Reply:
Done